helix-view/src/document.rs

```diff
@@ -96,6 +96,9 @@ pub struct Document {
     /// Current indent style.
     pub indent_style: IndentStyle,
 
+    /// An override of the language-specified tab width.
+    tab_width_override: Option<usize>,
```

Comment:
This override field feels a little awkward. Can we use `tab_width` instead and compute it from the `doc.language_config()` during initialization of the document - either inheriting from the editorconfig or doing what is currently the implementation of `Document::tab_width`? Then it should end up looking more like `indent_style` or `line_ending`